pennylane/ops/op_math/exp.py

```diff
@@ -245,19 +285,27 @@ def resource_params(self) -> dict:
 
     # pylint: disable=invalid-overridden-method, arguments-renamed
     @property
+    @queuing.QueuingManager.stop_recording()
     def has_decomposition(self):
         # TODO: Support nested sums in method
-        base = self.base
+        base = self.base.simplify()
         coeff = self.coeff
         if isinstance(base, SProd):
             coeff *= base.scalar
             base = base.base
-        is_pauli_rot = qml.pauli.is_pauli_word(self.base) and math.real(self.coeff) == 0
-        is_hamiltonian = isinstance(base, LinearCombination)
-        is_sum_of_pauli_words = isinstance(base, Sum) and all(
-            qml.pauli.is_pauli_word(o) for o in base
-        )
-        return is_pauli_rot or is_hamiltonian or is_sum_of_pauli_words
+
+        if self.num_steps is not None and isinstance(base, Sum) and base.is_hermitian:
+            return True
```

Comment:
Might not be a really accurate check for this branch, but `num_steps` is going away after the next release anyway.